packages/react/counter/src/Counter.tsx

```diff
@@ -39,79 +36,72 @@ export interface CounterProps extends React.ComponentProps<'span'> {
     filters?: FilterInput[]
     /** Propeller that the chart will respond to */
     propeller?: Propeller
+    /** Interval in milliseconds for refetching the data */
+    refetchInterval?: number
+    /** Whether to retry on errors. */
+    retry?: boolean
   }
   /** When true, shows a skeleton loader */
   loading?: boolean
 }
 
 export function Counter(props: CounterProps) {
-  const { value, query, prefixValue, sufixValue, styles, loading = false, localize, ...rest } = props
+  const {
+    value: staticValue,
+    query,
+    prefixValue,
+    sufixValue,
+    styles,
+    loading: isLoadingStatic = false,
+    localize,
+    ...rest
+  } = props
 
   /**
    * If the user passes `value` attribute, it
    * should behave as a static component without any graphql operation performed
    */
   const isStatic = !query
 
-  const [dataValue, setDataValue] = React.useState<string | null>()
   const [propsMismatch, setPropsMismatch] = React.useState(false)
-  const [hasError, setHasError] = React.useState(false)
-  const [isLoading, setIsLoading] = React.useState(false)
 
-  const filtersString = JSON.stringify(query?.filters || [])
   const counterRef = React.useRef<HTMLSpanElement>(null)
 
-  /**
-   * Fetches the counter data
-   * when the user doesn't provide
-   * its own `value`
-   */
-  const fetchData = React.useCallback(async () => {
-    try {
-      setIsLoading(true)
-      setHasError(false)
-
-      const filters = JSON.parse(filtersString)
-
-      const response = await request<CounterQuery, CounterQueryVariables>(
-        PROPEL_GRAPHQL_API_ENDPOINT,
-        CounterDocument,
-        {
-          counterInput: {
-            metricName: query?.metric,
-            timeRange: {
-              relative: query?.timeRange?.relative ?? null,
-              n: query?.timeRange?.n ?? null,
-              start: query?.timeRange?.start ?? null,
-              stop: query?.timeRange?.stop ?? null
-            },
-            filters,
-            propeller: query?.propeller
-          }
-        },
-        {
+  const {
+    isInitialLoading: isLoadingQuery,
```

Comment:
Here I changed to `isInitialLoading` because `isLoading` will always be `true` when the query is disabled, this is an intended behavior from react-query that means there is no data yet. To address this kind of cases they added `isInitialLoading` which is a derived flag from `isLoading && isFetching` so it should work for us.

https://tanstack.com/query/v4/docs/react/guides/disabling-queries#isinitialloading